automation/python-tests/test_fml.py

```diff
@@ -0,0 +1,112 @@
+from fml import FmlClient, FmlError, InternalError
+import json
+
+
+def test_instantiate_fml_client():
+    FmlClient('./automation/python-tests/resources/test.fml.yml', 'developer')
+
+
+def test_instantiate_fml_client_fails_if_invalid_yml():
+    err = None
+    try:
+        FmlClient('./automation/python-tests/resources/test-invalid.fml.yml', 'developer')
+    except (FmlError, InternalError) as e:
+        err = e
+
+    if err is None:
+        assert False, 'FmlClient instantiation did not throw an error for an invalid fml.yml'
+
+
+def test_instantiate_fml_client_fails_if_invalid_channel():
+    err = None
+    try:
+        FmlClient('./automation/python-tests/resources/test.fml.yml', 'release')
+    except (FmlError, InternalError) as e:
+        err = e
+
+    if err is None:
+        assert False, 'FmlClient instantiation did not throw an error for an invalid channel'
+
+
+def test_default_json():
+    client = FmlClient('./automation/python-tests/resources/test.fml.yml', 'developer')
+    defaults = json.loads(client.get_default_json())
+    assert defaults["example-feature"]["enabled"] is False
+    assert defaults["example-feature"]["something"] == "wicked"
+
+    client = FmlClient('./automation/python-tests/resources/test.fml.yml', 'nightly')
+    defaults = json.loads(client.get_default_json())
+    assert defaults["example-feature"]["enabled"] is True
+    assert defaults["example-feature"].get("something") is None
+
+
+def test_validate():
+    client = FmlClient('./automation/python-tests/resources/test.fml.yml', 'developer')
+    config = {
+        "featureId": "example-feature",
+        "value": {
+            "enabled": True
+        }
+    }
+    assert client.validate_feature_config(json.dumps(config)) is True
+
+
+def test_validate_false():
```

Comment:
Also test valid feature name and invalid value?

Reply:
Much better name, thanks!